CRM/Core/Resources.php

```diff
@@ -881,4 +882,27 @@ private function resolveFileName(&$fileName, $extName) {
     }
   }
 
+  /**
+   * @param string $url
+   * @return string
+   */
+  public function addCacheCode($url) {
+    parse_str(parse_url($url, PHP_URL_QUERY), $queryParts);
+    $existing = isset($queryParts['r']) ? $queryParts['r'] : NULL;
+    $latest = $this->cacheCode;
+
+    if ($existing) {
+      if ($existing === $latest) {
+        return $url; // no need to update
+      }
+      else {
+        return str_replace('r=' . $existing, 'r=' . $latest, $url);
```

Comment:
@totten It's my unfamiliarity with the system that led me to putting that there to cover all cases. I thought maybe it might be possible that the cache buster string could change. If you can be sure it'll never happen I can rewrite it without it to just cover

- $url has a query string and we need to append cache buster with '&'
- $url has no query string and we need to append cache buster with '?'

Which should simplify things a lot